### PR TITLE
Role Menu fixes

### PR DESCRIPTION
**Pull request by 1Revenger1**

### CODE REVIEW

File: src/main/java/com/wylxbot/wylx/Commands/Roles/RolesUtil/RoleMenu.java [hunk 1]
```diff
@@ -141,20 +144,34 @@ public void addReaction(RoleReaction newReaction) throws IllegalArgumentExceptio
...
+        try {
+            addReactionToMessage(newReaction.emoji());
+        } catch (ErrorResponseException e) {
+            if (e.getErrorResponse() == ErrorResponse.TOO_MANY_REACTIONS) {
+                throw new IllegalArgumentException("Discord has a max limit of 20 unique emojis per message. Please create a new menu or remove other roles from this menu.");
             } else {
-                throw new IllegalArgumentException("Emote does not exist");
+                throw new IllegalArgumentException(e.getMessage());
             }
         }
```
Comment (SnakeJ419) [inline src/main/java/com/wylxbot/wylx/Commands/Roles/RolesUtil/RoleMenu.java:155]: does
```
try {
    ...
} catch (Error1 e1) {
    ...
} catch (Error2 e2) {
    ...
}
```
work for this?

File: src/main/java/com/wylxbot/wylx/Commands/Roles/RolesUtil/RoleMenu.java [hunk 2]
```diff
@@ -6,13 +6,16 @@
...
 import net.dv8tion.jda.api.exceptions.ErrorResponseException;
+import net.dv8tion.jda.api.requests.ErrorResponse;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 
 public class RoleMenu {
 
+    private static int MAX_REACTIONS = 20; // Discord limit
```
Comment (SnakeJ419) [inline src/main/java/com/wylxbot/wylx/Commands/Roles/RolesUtil/RoleMenu.java:17]: does this get used anywhere?

**Review by SnakeJ419 — CHANGES_REQUESTED** [review]
Looks good, still needs the permissions change

File: src/main/java/com/wylxbot/wylx/Commands/Roles/RolesUtil/RoleMenu.java (hunk 1 above)
Comment (1Revenger1) [reply src/main/java/com/wylxbot/wylx/Commands/Roles/RolesUtil/RoleMenu.java:155]: I wanted to directly catch too many reactions, but I guess I could put that logic in the DMListener if that's what your asking.

Comment (SnakeJ419) [reply src/main/java/com/wylxbot/wylx/Commands/Roles/RolesUtil/RoleMenu.java:155]: Ah if you need to specficially catch the TOO_MANY_REACTIONS variant of ErrorResponseException then i think this is just fine

Comment (1Revenger1) [reply src/main/java/com/wylxbot/wylx/Commands/Roles/RolesUtil/RoleMenu.java:155]: Yeah, I wanted a better response than "Too many reactions" to the user lol.

File: src/main/java/com/wylxbot/wylx/Commands/Roles/ModifyRoleMenuCommand.java [hunk 3]
```diff
@@ -47,10 +47,14 @@ public void runCommand(CommandContext ctx) {
             return;
         }
 
+        if (!menu.getGuildID().equals(ctx.guildID())) {
+            ctx.event().getMessage().reply("Menu is from another server, please run ?modifyrolemenu from that server.").queue();
```
Comment (SnakeJ419) [inline src/main/java/com/wylxbot/wylx/Commands/Roles/ModifyRoleMenuCommand.java:51]: change it so its not a question mark

File: src/main/java/com/wylxbot/wylx/Commands/Roles/RolesUtil/DMListener.java [hunk 4]
```diff
@@ -38,15 +38,13 @@ private void resetTimer() {
...
+                DMListenerUserManager.getInstance().removeDMListener(user);
             }
         }, TIMEOUT);
     }
 
     public void quit() {
         timer.cancel();
-        user.getJDA().removeEventListener(this);
-        DMListenerUserManager.getInstance().removeDMListener(user);
     }
```
Comment (SnakeJ419) [inline src/main/java/com/wylxbot/wylx/Commands/Roles/RolesUtil/DMListener.java:48]: send a message saying the dm is being closed?

**Review by SnakeJ419 — CHANGES_REQUESTED** [review]

File: src/main/java/com/wylxbot/wylx/Commands/Roles/RolesUtil/DMListener.java (hunk 4 above)
Comment (1Revenger1) [reply src/main/java/com/wylxbot/wylx/Commands/Roles/RolesUtil/DMListener.java:48]: This is done in other places (Timer task and when "quit" is explicitly sent as a command). Only place it isn't done is when replacing one modify session with another. I felt it'd be confusing if there was a message saying that the menu was done being edited, then another prompt appeared. Got a good message to send if the session gets interrupted by another one?

Comment (SnakeJ419) [reply src/main/java/com/wylxbot/wylx/Commands/Roles/RolesUtil/DMListener.java:48]: "Existing session interrupted! Ending edit on <menu_id or menu_name> to start next session"

Comment (SnakeJ419) [reply src/main/java/com/wylxbot/wylx/Commands/Roles/RolesUtil/DMListener.java:48]: Feel like something should be said though, even if it probably will never matter

**Review by SnakeJ419 — APPROVED** [review]

**Pull request merged**